connect/src/com/telenor/connect/ui/ConnectWebViewClient.java

```diff
@@ -94,6 +109,70 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
     }
 
     @Override
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request) {
+        // Only implement interception logic when we have both cellular and WiFi interfaces active
+        if (ConnectSdk.getCellularNetwork() == null || ConnectSdk.getWifiNetwork() == null) {
+            return null;
+        }
+        if (shouldFetchThroughCellular(request.getUrl().toString())) {
+            return fetchUrlThroghCellular(request.getUrl().toString());
+        }
+        return null;
+    }
+
+    public boolean shouldFetchThroughCellular(String url) {
+        try {
```

Comment:
Yes. All except the one in the return clause. I think it's more readable this way.